src/qtui/topicwidget.cpp

```diff
@@ -152,13 +152,17 @@ void TopicWidget::setTopic(const QModelIndex& index)
         }
     }
 
-    _topic = sanitizeTopic(newtopic);
-    _readonly = readonly;
-
-    ui.topicEditButton->setVisible(!_readonly);
-    ui.topicLabel->setText(_topic);
-    ui.topicLineEdit->setPlainText(_topic);
-    switchPlain();
+    QString sanitizedNewTopic = sanitizeTopic(newtopic);
+    if (ui.stackedWidget->currentIndex() != 1 || sanitizedNewTopic != _topic)
```

Comment:
Not tested, but it might be fine to simply check if `_readonly != readonly` instead.  Reason being, if there's no changes to topic or readonly state, there shouldn't be any reason to set the topic, change the widget order, or set the edit button visibility.

```suggestion
    if (readonly != _readonly || sanitizedNewTopic != _topic)
```

(I'm surprised this function is even called on every new message; it feels like the `dataChanged` event is being too noisy.)

Reply:
I just tested this and it looks like it works.